src/core/crypto/storage.hpp

```diff
@@ -102,6 +105,7 @@ constexpr KeyRef kActiveDatasetNetworkKeyRef  = OPENTHREAD_CONFIG_PSA_ITS_NVM_OF
 constexpr KeyRef kActiveDatasetPskcRef        = OPENTHREAD_CONFIG_PSA_ITS_NVM_OFFSET + 4;
 constexpr KeyRef kPendingDatasetNetworkKeyRef = OPENTHREAD_CONFIG_PSA_ITS_NVM_OFFSET + 5;
 constexpr KeyRef kPendingDatasetPskcRef       = OPENTHREAD_CONFIG_PSA_ITS_NVM_OFFSET + 6;
+constexpr KeyRef kEcdsaRef                    = OPENTHREAD_CONFIG_PSA_ITS_NVM_OFFSET + 7;
```

Comment:
So this assumes that there's only one user of the `KeyPair` class but if another user is added at some point the two users will overwrite each other's data without a clear error. Wouldn't it be cleaner if this was `kSrpEcdsaRef` and was specified by the SRP client (e.g. `KeyPair` could take the key reference in the constructor)?

Reply:
Good point!! You are right, it would be nice to allow multiple users for ECDSA.

Reply:
Thanks!